src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -52,16 +54,21 @@ private async Task LoadAsync()
             var secrets = await _client.GetSecretsAsync(_vault).ConfigureAwait(false);
             do
             {
+                var tasks = new List<Task<KeyValuePair<string, string>>>();
                 foreach (var secretItem in secrets)
                 {
-                    if (!_manager.Load(secretItem) || (secretItem.Attributes?.Enabled != true))
+                    if (!_manager.Load(secretItem) || secretItem.Attributes?.Enabled != true)
                     {
                         continue;
                     }
+                    tasks.Add(LoadSecretValue(secretItem));
+                }
+
+                await Task.WhenAll(tasks);
 
-                    var value = await _client.GetSecretAsync(secretItem.Id).ConfigureAwait(false);
-                    var key = _manager.GetKey(value);
-                    data.Add(key, value.Value);
+                foreach (var task in tasks)
+                {
+                    data.Add(task.Result.Key, task.Result.Value);
```

Comment:
```suggestion
                    var (key, value) = task.Result;
                    data.Add(key, value);
```
?

Offtopic: it's a bit strange to "deconsruct" a `KeyValuePair<K, V>` if you want to add it to a dictionary :wink: